examples/question_answer/README.md

```diff
@@ -0,0 +1,144 @@
+- A remote instance of Astarte. We will assume you created your own realm on
+[Astarte Cloud](https://astarte.cloud/) and you will be managing it through the
+[Astarte Dashboard](https://github.com/astarte-platform/astarte-dashboard).
+- The [astartectl](https://github.com/astarte-platform/astartectl/releases) tool. This tool is an
+alternative to Astarte Dashboard to manage your remote Astarte instance. It includes some additional
+functionality used in this example.
+
+**NB.** This SDK does NOT support ESP-IDF version 5. Earlier versions 4 and 3 are supported.
+
+## Installing the interfaces on Astarte
+
+We shall start by installing the required interfaces on the remote Astarte instance.
+
+To install a new interface using the Astarte Dashboard follow this sequence of steps:
+1. In the *Interfaces* tab click on *Install a new interface ...*
+2. In the interface editor fill in the required information for your interface. If you already have
+a interface definition in the `.json` format you can paste that directly in the editor at the
+right of the page.
+3. Finish by clicking on install interface
+4. Refresh the page to check that the interface has been installed correctly.
+
+Install two interfaces using these two `.json` files:
+- `interfaces/org.astarteplatform.esp32.DeviceDatastream.json`
+- `interfaces/org.astarteplatform.esp32.ServerDatastream.json`
+
+## Manually registering a new device on Astarte
+
+Next, let's register a new device on the remote Astarte instance.
+
+The following procedure shows how to do this:
+1. In the *Devices* tab click on *Register a new device*
+2. Click *Generate random ID* and then on *Register device*
+3. Copy the device credential secret somewhere local
+
+**NB**: While the device ID can always be recovered from the Astarte Dashboard, the
+credential secred is only shown once during the device registration procedure.
+Take care of storing it somewhere local as it will be required in the next
+step.
+
+## Configuring the SDK
+
+We will assume you are running the ESP-IDF from command line. If you are using the GUIs provided
+with the Visual Studio Code or the Eclipse plugins then the configuration procedure might be
+slightly different.
+
+Open the configuration tool using the following command:
+| ESP-IDF v4.x  | ESP-IDF v3.x |
+| ------------- | ------------ |
+| `idf.py menuconfig` | `make menuconfig` |
+
+In the *Astarte question/answer example* tab fill in:
+- *WiFi SSID* the SSID of your wifi network
+- *WiFi Password* the SSID of your wifi network
+- *Credentials secret* the device specific credential secret generated in the previous step
+- *Device hardware ID* the device hardware ID generated in the previous step
+
+In the *Component config ---> Astarte SDK* tab fill in:
+- *Astarte realm* place here the name of your Astarte realm
+- *Astarte pairing base URL* use the Base API URL for your remote Astarte instance adding at the
+end `/pairing`
```

Comment:
Since you are referencing to Astarte Cloud throughout the document, I think we could reference to the actual URL. Something like:
```suggestion
- *Astarte pairing URL* use the Base API URL for your remote Astarte instance adding at the
end `/pairing` (e.g.: if you are using Astarte Cloud: https://api.eu1.astarte.cloud/pairing).
```

Reply:
I agree with the changes, then only addition is that removing the `base` at the beginning of the sentence only makes sense if we also change the name of the IDF menu it's referring to.
So I will also add that in my changes.

Reply:
I think this would propagate also in some of the sources.
For example in the `astarte_pairing_config` struct, and the `CONFIG_ASTARTE_PAIRING_URL` macro.
I am not sure if this should be bundled in this PR or should be done in its own PR.